### PR TITLE
Add possibility for a honeypot field for 1.x

### PR DESCRIPTION
**Pull request by Quehnie**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes (copy from version for Sulu 2)
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Also add the honeypot field possibility for this older version.

#### Why?

Spam problems.

#### Example Usage

In documentation

### CODE REVIEW

File: Form/Type/DynamicFormType.php [hunk 1]
```diff
@@ -149,6 +157,21 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+        if ($this->honeyPotField) {
+            $builder->add(
+                str_replace(' ', '_', strtolower($this->honeyPotField)),
+                EmailType::class,
+                [
+                    'label' => $this->honeyPotField,
+                    'mapped' => false,
+                    'attr' => [
+                        'honeypot' => true
+                    ],
```
Comment (alexander-schranz) [inline Form/Type/DynamicFormType.php:169]: attribute should not be here.

Instead it should be:

```php
'block_name' => 'honeypot',
```

See https://symfony.com/doc/3.4/reference/forms/types/form.html#block-name

The block_name should as in the docs be then `_honeypot_row`.

File: Resources/doc/dynamic.md [hunk 2]
```diff
@@ -163,6 +163,63 @@ php bin/adminconsole debug:container --tag=sulu_form.dynamic.type
...
+```
+
+And then hide it in your css with:
+
+```css
+.honung {
+    display: none;
+}
+```
+
```
Comment (alexander-schranz) [inline Resources/doc/dynamic.md:221]: This should be the same as here: https://github.com/sulu/SuluFormBundle/blob/master/Resources/doc/dynamic.md#style-honey-pot-field

The block name should be something like `_honeypot_row`

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: Form/Type/DynamicFormType.php [hunk 3]
```diff
@@ -149,6 +157,19 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
         ]);
 
+        if ($this->honeyPotField) {
+            $builder->add(
+                str_replace(' ', '_', strtolower($this->honeyPotField)),
+                EmailType::class,
+                [
+                    'label' => $this->honeyPotField,
+                    'mapped' => false,
+                    'block_name' => 'honeypot',
```
Comment (alexander-schranz) [inline Form/Type/DynamicFormType.php:167]: `block _honeypot_row` didn't work? then this can be removed

File: Form/Handler.php [hunk 4]
```diff
@@ -71,14 +81,18 @@ public function __construct(
         Mail\HelperInterface $mailHelper,
         EngineInterface $templating,
         EventDispatcherInterface $eventDispatcher,
-        MediaManager $mediaManager
+        MediaManager $mediaManager,
+        string $honeyPotStrategy = self::HONEY_POT_STRATEGY_SPAM,
+        string $honeyPotField = null
```
Comment (alexander-schranz) [inline Form/Handler.php:86]: PHP Docs missing:

```
     * @param string $honeyPotStrategy
     * @param string $honeyPotField
```

File: Form/Handler.php [hunk 5]
```diff
@@ -71,14 +81,18 @@ public function __construct(
         Mail\HelperInterface $mailHelper,
         EngineInterface $templating,
         EventDispatcherInterface $eventDispatcher,
-        MediaManager $mediaManager
+        MediaManager $mediaManager,
+        string $honeyPotStrategy = self::HONEY_POT_STRATEGY_SPAM,
```
Comment (alexander-schranz) [inline Form/Handler.php:85]: `string` can not be used in all supported php versions and need to be removed

File: Form/Handler.php [hunk 6]
```diff
@@ -155,8 +186,9 @@ private function sendMails(FormInterface $form, FormConfigurationInterface $conf
      *
      * @param FormInterface $form
      * @param MailConfigurationInterface $configuration
+     * @param string $subjectPrefix
      */
-    private function sendMail(FormInterface $form, MailConfigurationInterface $configuration)
+    private function sendMail(FormInterface $form, MailConfigurationInterface $configuration, string $subjectPrefix = '')
```
Comment (alexander-schranz) [inline Form/Handler.php:191]: `string` typehint can not be used in this PHP version and need to be removed.

File: Form/Handler.php [hunk 7]
```diff
@@ -142,7 +167,13 @@ private function save(FormInterface $form, FormConfigurationInterface $configura
     private function sendMails(FormInterface $form, FormConfigurationInterface $configuration)
     {
         if ($adminMailConfiguration = $configuration->getAdminMailConfiguration()) {
-            $this->sendMail($form, $adminMailConfiguration);
+            $subjectPrefix = '';
+
```
Comment (alexander-schranz) [inline Form/Handler.php:171]: remove this empty (avoids us conflicts when merging into 2.x branch)

File: Form/Handler.php [hunk 8]
```diff
@@ -303,4 +341,25 @@ protected function getPlainText(FormInterface $form, MailConfigurationInterface
             )
         );
     }
+
```
Comment (alexander-schranz) [inline Form/Handler.php:344]: in the merge it looks like this line contains spaces remove the spaces if that is the case.

File: Resources/doc/dynamic.md [hunk 9]
```diff
@@ -163,6 +163,52 @@ php bin/adminconsole debug:container --tag=sulu_form.dynamic.type
...
+
```
Comment (alexander-schranz) [inline Resources/doc/dynamic.md:211]: Only one empty line should be here.

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Some small changes.

File: Form/Type/DynamicFormType.php (hunk 3 above)
Comment (Quehnie) [reply Form/Type/DynamicFormType.php:167]: it does work but with prefix of the form so _dynamic_honeypot_row

Comment (alexander-schranz) [reply Form/Type/DynamicFormType.php:167]: okay

File: Form/Type/DynamicFormType.php [hunk 10]
```diff
@@ -42,10 +48,12 @@ class DynamicFormType extends AbstractType
      */
     public function __construct(
         FormFieldTypePool $typePool,
-        Checksum $checksum
+        Checksum $checksum,
+        ?string $honeyPotField = null
```
Comment (alexander-schranz) [inline Form/Type/DynamicFormType.php:52]: This typehint should also be removed and `string|null $honeyPotField` added to the construct function

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Pull request merged**